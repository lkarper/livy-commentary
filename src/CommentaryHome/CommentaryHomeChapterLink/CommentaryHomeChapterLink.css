.CommentaryHomeChapterLink__a {
    color: black;
    text-decoration: none;
}

.CommentaryHomeChapterLink__a:hover, .CommentaryHomeChapterLink__a:focus {
    text-decoration: underline;
}