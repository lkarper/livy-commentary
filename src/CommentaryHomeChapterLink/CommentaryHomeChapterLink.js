import React from 'react';
import { Link } from 'react-router-dom';
import CommentaryHomeSectionLink from '../CommentaryHomeSectionLink/CommentaryHomeSectionLink';

const CommentaryHomeChapterLink = (props) => {
    const { chapter } = props;

    return (
        <div>
            <Link>Chapter {chapter.chapter_number}: {chapter.chapter_title}</Link>
            {chapter.sections.map(section => <CommentaryHomeSectionLink key={section.section_number} section={section} />)}
        </div>
    );
}

export default CommentaryHomeChapterLink;