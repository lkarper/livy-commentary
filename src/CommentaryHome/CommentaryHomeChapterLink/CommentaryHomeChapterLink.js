import React from 'react';
import { Link } from 'react-router-dom';
import CommentaryHomeSectionLink from '../CommentaryHomeSectionLink/CommentaryHomeSectionLink';
import './CommentaryHomeChapterLink.css';

const CommentaryHomeChapterLink = (props) => {
    const { chapter, book_number } = props;

    return (
        <div className='CommentaryHomeChapterLink__chapter-link-container'>
            <h3>
                <Link 
                    to={`/commentary-read?book=${book_number}&chapter=${chapter.chapter_number}`}
                >
                    Chapter {chapter.chapter_number}: {chapter.chapter_title}
                </Link>
            </h3>
            <ul>
                {chapter.sections.map(section => 
                    <CommentaryHomeSectionLink 
                        key={section.section_number} 
                        book_number={book_number} 
                        chapter_number={chapter.chapter_number} 
                        section={section} 
                    />)
                }
            </ul>
        </div>
    );
}

export default CommentaryHomeChapterLink;