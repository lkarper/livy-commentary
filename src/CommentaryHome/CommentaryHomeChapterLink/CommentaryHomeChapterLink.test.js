import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CommentaryHomeChapterLink from './CommentaryHomeChapterLink';

describe('CommentaryHomeChapterLink component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <CommentaryHomeChapterLink />
            </BrowserRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <CommentaryHomeChapterLink />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
