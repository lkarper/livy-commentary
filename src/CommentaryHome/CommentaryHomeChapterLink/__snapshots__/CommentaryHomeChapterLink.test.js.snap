// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentaryHomeChapterLink component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CommentaryHomeChapterLink
      book_number={0}
      chapter={
        Object {
          "chapter_number": 0,
          "sections": Array [],
        }
      }
    >
      <div
        className="CommentaryHomeChapterLink__chapter-link-container"
      >
        <h3>
          Error
        </h3>
        <p>
          Looks like something went wrong. Check your connection and try again.
        </p>
      </div>
    </CommentaryHomeChapterLink>
  </Router>
</BrowserRouter>
`;
